app/controllers/attachment_references_controller.rb

```diff
@@ -0,0 +1,14 @@
+class AttachmentReferencesController < ApplicationController
+  load_resource :attachment_reference
+  before_action :authorize_resource
+
+  def show
+    redirect_to @attachment_reference.url
+  end
+
+  private
+
+  def authorize_resource
+    authorize!(:read, @attachment_reference.attachable)
```

Comment:
but now we don't authorize showing the attachable item right?

Actually, maybe that's necessary, the more I think about it...